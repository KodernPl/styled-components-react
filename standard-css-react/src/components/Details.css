@import '../styles/variables.css';

.details {
  list-style: none;
  padding: 0 4rem;
  margin: 4rem auto .6rem;

  &__item {
    text-align: center;
    font-size: 1.4rem;
    padding: 1rem;
    color: var(--color-dark-grey);

    &:not(:last-child) {
      border-bottom: 1px solid var(--color-alpha-black);
    }
  }
}
