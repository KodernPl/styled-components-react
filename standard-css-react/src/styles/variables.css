:root {
  --color-white: white;
  --color-gradient1: #667db6;
  --color-gradient2: #0082c8;
  --color-gradient3: #0082c8;
  --color-gradient4: #667db6;
  --color-white: #fff;
  --color-black: #000;
  --color-dark-grey: #222;
  --color-darkest-grey: #111;
  --color-grey: grey;
  --color-palevioletred: palevioletred;
  --color-sc-pink: rgb(219, 112, 147);
  --color-sc-orange: rgb(218, 163, 87);
  --color-alpha-black: rgba(0, 0, 0, 0.2);
  --color-alpha-grey: rgba(20, 20, 20, 0.1);
  --gradient-light: linear-gradient(to top right, rgb(219, 112, 147), rgb(218, 163, 87));
  --gradient-dark: linear-gradient(to top right, #667db6, #0082c8, #0082c8);
}

@custom-media --desktop (min-width: 1024px);
