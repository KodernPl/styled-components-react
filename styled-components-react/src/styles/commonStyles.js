import styled from 'styled-components';

export const CenterElements = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
`;

export const CustomLink = styled.a`
  display: inline-block;
  width: 20rem;
  padding: .5rem 0;
  margin: .5rem 1rem;
  border-width: .2rem;
  border-radius: .3rem;
  border-style: solid;
  border-image: initial;
  cursor: pointer;
  text-align: center;
  text-decoration: none;
  font-size: 1.2rem;
`;
