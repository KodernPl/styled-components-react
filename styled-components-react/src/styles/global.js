import { createGlobalStyle } from 'styled-components';

export default createGlobalStyle`
  html {
    box-sizing: border-box;
    font-size: 62.5%;
    font-family: sans-serif;
  }

  body {
    margin: 0;
    padding: 0;
  }

  *,
  *::before,
  *::after {
    box-sizing: inherit;
  }
`;
