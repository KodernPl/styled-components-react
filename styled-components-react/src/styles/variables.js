export const COLORS = {
  gradient1: '#667db6',
  gradient2: '#0082c8',
  gradient3: '#0082c8',
  gradient4: '#667db6',

  white: '#fff',
  black: '#000',
  darkGrey: '#222',
  darkestGrey: '#111',

  grey: 'grey',
  palevioletred: 'palevioletred',

  scPink: 'rgb(219, 112, 147)',
  scOrange: 'rgb(218, 163, 87)',

  alphaBlack: 'rgba(0, 0, 0, 0.2)',
  alphaGrey: 'rgba(20, 20, 20, 0.1)',
};

export const THEMES = {
  light: 'light',
  dark: 'dark',
};

export const GRADIENTS = {
  light: `linear-gradient(to top right, ${COLORS.scPink}, ${COLORS.scOrange})`,
  dark: `linear-gradient(to top right, ${COLORS.gradient1}, ${COLORS.gradient2}, ${COLORS.gradient4})`,
};

export const BREAKPOINTS = {
  desktop: '(min-width: 1024px)',
};
