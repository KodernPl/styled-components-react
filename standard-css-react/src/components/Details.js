import React from 'react';
import PropTypes from 'prop-types';

import './Details.css';

const ns = 'details';

const displayName = 'Details';

const propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

const Details = ({ children }) => (
  <ul className={ns}>
    {children.map(child => <li className={`${ns}__item`} key={child.props.children}>{child}</li>)}
  </ul>
);

Details.displayName = displayName;
Details.propTypes = propTypes;

export default Details;
