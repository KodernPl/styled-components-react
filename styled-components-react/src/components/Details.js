import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { COLORS } from '../styles/variables';

const DetailsWrapper = styled.ul`
  list-style: none;
  padding: 0 4rem;
  margin: 4rem auto .6rem;
`;

const Item = styled.li`
  text-align: center;
  font-size: 1.4rem;
  padding: 1rem;
  color: ${COLORS.darkGrey};

  &:not(:last-child) {
    border-bottom: 1px solid ${COLORS.alphaBlack};
  }
`;

const displayName = 'Details';

const propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

const Details = ({ children }) => (
  <DetailsWrapper>
    {children.map(child => <Item key={child.props.children}>{child}</Item>)}
  </DetailsWrapper>
);

Details.displayName = displayName;
Details.propTypes = propTypes;

export default Details;
