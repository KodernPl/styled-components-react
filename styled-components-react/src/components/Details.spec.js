import React from 'react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';

import { THEMES } from '../styles/variables';

import Details from './Details';

const mockComponent = theme => (
  <ThemeProvider theme={{ mode: theme }}>
    <Details>
      <span>TEST</span>
      <span>TEST2</span>
      <span>TEST3</span>
    </Details>
  </ThemeProvider>
);
[
  THEMES.light,
  THEMES.dark,
].forEach((theme) => {
  const cmp = renderer.create(mockComponent(theme)).toJSON();

  describe(`${theme} theme`, () => {
    test('renders correctly child elements ', () => {
      expect(cmp).toMatchSnapshot();
      expect(cmp.children.length).toEqual(3);
    });

    test('should list item have correct font size', () => {
      expect(cmp.children[0]).toHaveStyleRule('font-size', '1.4rem');
    });
  });
});
