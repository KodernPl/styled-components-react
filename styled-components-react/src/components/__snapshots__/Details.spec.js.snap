// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dark theme renders correctly child elements  1`] = `
.c0 {
  list-style: none;
  padding: 0 4rem;
  margin: 4rem auto .6rem;
}

.c1 {
  text-align: center;
  font-size: 1.4rem;
  padding: 1rem;
  color: #222;
}

.c1:not(:last-child) {
  border-bottom: 1px solid rgba(0,0,0,0.2);
}

<ul
  className="c0"
>
  <li
    className="c1"
  >
    <span>
      TEST
    </span>
  </li>
  <li
    className="c1"
  >
    <span>
      TEST2
    </span>
  </li>
  <li
    className="c1"
  >
    <span>
      TEST3
    </span>
  </li>
</ul>
`;

exports[`light theme renders correctly child elements  1`] = `
.c0 {
  list-style: none;
  padding: 0 4rem;
  margin: 4rem auto .6rem;
}

.c1 {
  text-align: center;
  font-size: 1.4rem;
  padding: 1rem;
  color: #222;
}

.c1:not(:last-child) {
  border-bottom: 1px solid rgba(0,0,0,0.2);
}

<ul
  className="c0"
>
  <li
    className="c1"
  >
    <span>
      TEST
    </span>
  </li>
  <li
    className="c1"
  >
    <span>
      TEST2
    </span>
  </li>
  <li
    className="c1"
  >
    <span>
      TEST3
    </span>
  </li>
</ul>
`;
